### PR TITLE
feat(mcdu): page format helper

### PR DESCRIPTION
**Pull request by derl30n**
## Summary of Changes

This pr aims to ease page formatting by anchoring "data" to a specific location.
Usage can be on a single mcdu line or the entire page. No integration for existing pages required.
Intended to be used when re-working pages etc. although just converting pages for cosmetics sounds just fine.

### Features:
- Anchor text to a fixed position
- Align text to either left or right
   (examples can be read from left to write)
   - with left align -> anchor is at the first char of the text (at the start of the text)
   - with right align -> anchor is at the last char of the text (at the end of the text)
- Text size can be changed dynamically
- Color can be changed dynamically

#### Default values for attributes are:
- alignment: left
- color: white
- size: normal

### Example of MCDU Menu Page included in this pr
| Before | After |
| --- | --- |
| image | image |

### Example of MCDU Departures Page not included in this pr
| Before | After |
| --- | --- |
| image | image |

## Additional context

This pr aims to provide a bridge until the overhauled page system is in place.

Discord username (if different from GitHub): DerL30N#3751

## Testing instructions

Observe mcdu page looks the same as described in the picture.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by 2hwk — APPROVED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/**
+ * Used to displayed data on a mcdu page when using the formatting helper
+ * @param index {integer} valid range from 0 to 23
+ * @param text {str} text to be displayed
+ * @param att {list} attributes of the text, e.g. text size, color and/or alignment
+ */
+class Column {
+    constructor(index, text, ...att) {
```
Comment (ErickSharp) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js:8]:
```suggestion
/** Used to displayed data on a mcdu page when using the formatting helper */
class Column {
    /**
     * @param {number} index - valid range from 0 to 23
     * @param {string} text - text to be displayed
     * @param {...({ color: string } | { size: string } | { align: bool })} att - attributes of the text, e.g. text size, color and/or alignment
     */
    constructor(index, text, ...att) {
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+Column.small = { "size": "small" };
+Column.big = { "size": "big" };
+Column.amber = { "color": "amber"};
+Column.red = { "color": "red"};
+Column.green = { "color": "green"};
+Column.cyan = { "color": "cyan"};
+Column.white = { "color": "white"};
+Column.magenta = { "color": "magenta"};
+Column.yellow = { "color": "yellow"};
+Column.inop = { "color": "inop"};
```
Comment (ErickSharp) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js:32]:
```suggestion
Column.amber = { "color": "amber" };
Column.red = { "color": "red" };
Column.green = { "color": "green" };
Column.cyan = { "color": "cyan" };
Column.white = { "color": "white" };
Column.magenta = { "color": "magenta" };
Column.yellow = { "color": "yellow" };
Column.inop = { "color": "inop" };
```

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+/**
+ * Returns a formatted mcdu page template
+ * @param lines {array[Column[]]} mcdu lines
+ * @returns {string[]} mcdu template
+ */
+const FormatTemplate = lines => lines.map(line => FormatLine(...line));
+
+/**
+ * Returns a formatted mcdu line
+ * @param columns {Column}
```
Comment (ErickSharp) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js:43]:
```suggestion
 * @param columns {@link Column}
```

**Review by ErickSharp — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+Column.green = { "color": "green"};
+Column.cyan = { "color": "cyan"};
+Column.white = { "color": "white"};
+Column.magenta = { "color": "magenta"};
+Column.yellow = { "color": "yellow"};
+Column.inop = { "color": "inop"};
+
+/**
+ * Returns a formatted mcdu page template
+ * @param lines {array[Column[]]} mcdu lines
```
Comment (ErickSharp) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_Format.js:36]:
```suggestion
 * @param {Column[][]} lines - mcdu lines
```

**Review by ErickSharp — CHANGES_REQUESTED** [review]

**Review by ErickSharp — APPROVED** [review]
Awesome 🤗🤗

**Comment by alexr4339** [comment]
Quality Assurance Tester Report

Discord : Alex R#6663
Object of testing: PR#7294
Tier of Testing : 1
Date : 20/06/2022

Testing Process:
Loaded A/C on runway and checked whether all MCDU pages look the same as before and dont have any errors or misaligned elements. (To me both pictures given above, before and after, look the same to me. Thats why I decided to check whether all pages look as intended)

Negatives:
N/A

Testing Results:
Passed

**Pull request merged**